components/agreements/AgreementWithActions.tsx

```diff
@@ -69,7 +69,7 @@ export const AgreementWithActions = ({ agreement, onEdit, onDelete }) => {
   const intl = useIntl();
   return (
     <React.Fragment>
-      <Agreement agreement={agreement} />
+      <Agreement agreement={agreement} openFileViewer={() => openFileViewer(agreement.attachment)} />
```

Comment:
I haven't tested it, but it seems the `Agreement.openFileViewer` prop will be defined even if `AgreementWithActions.openFileViewer` is undefined.

Reply:
Indeed! It still works since `openFileViewer` is always defined, but the types don't reflect that appropriately. I'll update